### ISSUE TITLE
Deployment failed due to regression of #606

### ISSUE THREAD

**Issue by michaelarnauts** [forcedotcom/source-deploy-retrieve#611]
I'm unable to deploy since the last release of this library.

The `rest` seems to be missing in  `apiOptions`. This has been added in #606 .

I'm running on a clean env in CI, so I have downloaded all latest dependency packages.

This is some of the output of the `force:source:deploy` with a `--dev-debug`:
```
2022-04-19T13:07:14.866Z sfdx init version: @oclif/command@1.8.16 argv: [ 'force:source:deploy', '--wait', '20', '--sourcepath', 'force-app', '--targetusername', 'org', '--testlevel=RunLocalTests', '--loglevel=trace', '--verbose' ]
2022-04-19T13:07:14.869Z @oclif/config runCommand force:source:deploy [ '--wait', '20', '--sourcepath', 'force-app', '--targetusername', 'org', '--testlevel=RunLocalTests', '--loglevel=trace', '--verbose' ]
2022-04-19T13:07:14.869Z @oclif/config:@salesforce/plugin-source require /usr/lib/node_modules/sfdx-cli/node_modules/@salesforce/plugin-source/lib/commands/force/source/deploy.js
2022-04-19T13:07:14.930Z sfdx:core TRACE Setup child 'core:config' logger instance
/usr/lib/node_modules/sfdx-cli/node_modules/@salesforce/source-deploy-retrieve/lib/src/client/metadataApiDeploy.js:211
        this.isRestDeploy = !!options.apiOptions?.rest;
                                                 ^
SyntaxError: Unexpected token '.'
    at new Script (vm.js:88:7)
    at NativeCompileCache._moduleCompile (/usr/lib/node_modules/sfdx-cli/node_modules/v8-compile-cache/v8-compile-cache.js:240:18)
    at Module._compile (/usr/lib/node_modules/sfdx-cli/node_modules/v8-compile-cache/v8-compile-cache.js:184:36)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Module._load (internal/modules/cjs/loader.js:708:14)
    at Function._load (/usr/lib/node_modules/sfdx-cli/node_modules/@salesforce/lazy-require/lib/LazyLoader.js:99:24)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (/usr/lib/node_modules/sfdx-cli/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (/usr/lib/node_modules/sfdx-cli/node_modules/@salesforce/source-deploy-retrieve/lib/src/client/index.js:10:27)
2022-04-19T13:07:15.031Z lazy-require saving type cache to /root/.cache/sfdx/module-types.json
2022-04-19T13:07:15.031Z lazy-require saved type cache
```

**Comment by shetzel** [comment]
I'm not aware of any changes to the node version for the CLI or plugins.  Looks like node v12 is EOL at the end of this month anyway so bumping it is probably warranted.

**Comment by shetzel** [comment]
@michaelarnauts - the regression is [here](https://github.com/forcedotcom/source-deploy-retrieve/pull/610/files#diff-c0d527bf7b08c376a02d793f515e4ae3ae5cdda654c6e66a75f8a99927fd23a8R8), although with node v12 being EOL in 10 days we'd like to just keep it and bump engine versions.

**Comment by michaelarnauts** [comment]
I've upgraded my version of node, so all is fine now. 

It took a while to find out what the issue was, since all I got was the message "SyntaxError: Unexpected token '.'", without a stacktrace or any context. And untill a few days, I could deploy fine without an issue.

It would make sense to display an error stating Node 12 is not supported or something. That would make troubleshooting a lot easier :)
